btrace-core/src/main/java/org/openjdk/btrace/core/BTraceUtils.java

```diff
@@ -3040,6 +3041,21 @@ private static Field getField(Class<?> clazz, String name, boolean throwError) {
                 field.setAccessible(true);
                 return field;
               } catch (Exception exp) {
+                if(exp instanceof NoSuchFieldException) {
+                  Class superClass = clazz.getSuperclass();
+                  Field field = null;
+                  while (Objects.isNull(field) && Objects.nonNull(superClass)) {
+                    try {
+                      field = superClass.getDeclaredField(name);
+                    } catch (NoSuchFieldException e) {
+                      // Ignore exception and keep looking up
+                    }
+                  }
```

Comment:
This whole loop can be moved up and unified with the original code @ L3040-3042 - such that you walk up the class hierarchy until you find the field. If you are at `java.lang.Object` and still have no resolved field you should rethrow the first `NoSuchFieldException`.

Also, you don't seem to be actually walking up the class hierarchy - I would expect a line like `superClass = superClass.getSuperClass()` somewhere inside the loop.